provisioner/windows-restart/provisioner.go

```diff
@@ -174,28 +174,39 @@ WaitLoop:
 }
 
 var waitForCommunicator = func(p *Provisioner) error {
+	runCustomRestartCheck := true
+	if p.config.RestartCheckCommand == DefaultRestartCheckCommand {
+		runCustomRestartCheck = false
+	}
+	// this is the user configurable command
+	cmdRestartCheck := &packer.RemoteCmd{Command: p.config.RestartCheckCommand}
+	log.Printf("Checking that communicator is connected with: '%s'",
+		cmdRestartCheck.Command)
 	for {
-		cmd := &packer.RemoteCmd{Command: p.config.RestartCheckCommand}
-		var buf, buf2 bytes.Buffer
-		cmd.Stdout = &buf
-		cmd.Stdout = io.MultiWriter(cmd.Stdout, &buf2)
 		select {
 		case <-p.cancel:
 			log.Println("Communicator wait canceled, exiting loop")
 			return fmt.Errorf("Communicator wait canceled")
 		case <-time.After(retryableSleep):
 		}
-
-		log.Printf("Checking that communicator is connected with: '%s'", cmd.Command)
-
-		err := cmd.StartWithUi(p.comm, p.ui)
-
-		if err != nil {
-			log.Printf("Communication connection err: %s", err)
-			continue
+		if runCustomRestartCheck == true {
+			// run user-configured restart check
+			err := cmdRestartCheck.StartWithUi(p.comm, p.ui)
+			if err != nil {
+				log.Printf("Communication connection err: %s", err)
+				continue
+			}
+			log.Printf("Connected to machine")
+			runCustomRestartCheck = false
 		}
+		// this is the non-user-configurable check that powershell
+		// modules have loaded
+		cmdModuleLoad := &packer.RemoteCmd{Command: DefaultRestartCheckCommand}
```

Comment:
I think you probably want something like this here:
```
	var stdout bytes.Buffer
	cmd := &packer.RemoteCmd{
		Command: DefaultRestartCheckCommand,
		Stdout:  &stdout
	}

	if err := comm.Start(cmd); err != nil {
		err = fmt.Errorf("Failed blah blah blah: %s", err)
                return err
	}
```

Reply:
as opposed to doing the start with ui thing below?

Reply:
yeah, the startwithui thing outputs the remote command to the user, as if it were running locally. We really don't care about the output here, since we're just checking it ourselves, right?

Reply:
I guess; this is how the restart check was originally written so I didn't question it.  I'm happy to change it though.